api/src/test/java/jakarta/data/metamodel/restrict/BasicRestrictionRecordTest.java

```diff
@@ -23,109 +23,236 @@
 import jakarta.data.mock.entity._Book;
 
 import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
+import java.util.Set;
+
 import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
 
 
+@SuppressWarnings("unchecked")
```

Comment:
@njr-11 I put it on the whole class, could you check it now?

Reply:
That works.